pallets/asset-manager/src/lib.rs

```diff
@@ -426,8 +426,20 @@ pub mod pallet {
                 AssetIdLocation::<T>::contains_key(&asset_id),
                 Error::<T>::UpdateNonExistAsset
             );
+
+            ensure!(
```

Comment:
Perhaps question mark `?` is more elegant.

Reply:
Seems you reverted `?`.

Reply:
Ah yes, I thought this was the cause of the other problem and reverted. But as it turns out it's not the problem so unreverting it now :D